molecule/default/verify.yml

```diff
@@ -225,148 +226,155 @@
-        - RekeyLimit {{ sshd_rekey_limit }}
-        - StrictModes {{ sshd_strict_modes }}
-        - Subsystem {{ sshd_subsystem }}
-        - TCPKeepAlive {{ sshd_tcp_keep_alive }}
-        - UseDNS {{ sshd_use_dns }}
-        - UsePAM {{ sshd_use_pam }}
-        - X11Forwarding {{ sshd_x11_forwarding }}
+      loop: "{{ sshd_config_parameters }}"
       when: sshd_config_directory.stat.exists
 
     - name: Verify sshd runtime configuration
       become: true
       environment:
         PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
       ansible.builtin.command: sshd -T
+      check_mode: false
       register: sshd_config
       changed_when: false
-      failed_when: item not in sshd_config.stdout_lines
-      with_items:
-        - allowgroups sudo
-        - allowgroups vagrant
-        - allowagentforwarding {{ sshd_allow_agent_forwarding }}
-        - allowtcpforwarding {{ sshd_allow_tcp_forwarding }}
-        - authenticationmethods {{ sshd_authentication_methods }}
-        - banner {{ sshd_banner }}
-        - clientalivecountmax {{ sshd_client_alive_count_max | int }}
-        - clientaliveinterval {{ sshd_client_alive_interval | int }}
-        - compression {{ sshd_compression }}
-        - logingracetime {{ sshd_login_grace_time | int }}
-        - maxauthtries {{ sshd_max_auth_tries | int }}
-        - maxsessions {{ sshd_max_sessions | int }}
-        - maxstartups {{ sshd_max_startups }}
-        - permitrootlogin {{ sshd_permit_root_login }}
-        - port {{ sshd_port | int }}
-        - subsystem {{ sshd_subsystem }}
-        - usedns {{ sshd_use_dns }}
-        - usepam {{ sshd_use_pam }}
-        - x11forwarding {{ sshd_x11_forwarding }}
-
-    - name: Verify sshd runtime ciphers and algorithms
-      become: true
-      environment:
-        PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
-      ansible.builtin.command: sshd -T
-      register: sshd_algo_runtime
-      changed_when: false
-      failed_when: item not in sshd_algo_runtime.stdout_lines
-      when: not crypto_policies_config or not set_crypto_policy
-      with_items:
-        - casignaturealgorithms {{ sshd_ca_signature_algorithms | replace(', ', ',') }}
-        - ciphers {{ sshd_ciphers | replace(', ', ',') }}
-        - hostkeyalgorithms {{ sshd_host_key_algorithms | replace(', ', ',') }}
-        - kexalgorithms {{ sshd_kex_algorithms | replace(', ', ',') }}
-        - macs {{ sshd_macs | replace(', ', ',') }}
-
-    - name: Verify sshd RequiredRSASize
-      become: true
-      ansible.builtin.lineinfile:
-        dest: /etc/ssh/sshd_config.d/01-hardening.conf
-        line: RequiredRSASize {{ sshd_required_rsa_size }}
-        state: present
-      check_mode: true
-      register: sshd_config_rsasize
-      failed_when: sshd_config_rsasize is changed
-      when: sshd_config_directory.stat.exists and ssh_installed_version is version('9.1', '>')
+      failed_when: >-
```

Comment:
This one needs a fix.

Reply:
Do you have more details about the error? I ran a targeted test on an Ubuntu instance, and didn't detect any errors.

Reply:
It's the jinja regex and curly brackets that's causing issues, I believe https://ansible.readthedocs.io/projects/lint/rules/no-jinja-when/ but I'll post proper details later on

Reply:
Can you generate more details about the error?